bika/lims/browser/invoicebatch.py

```diff
@@ -68,56 +98,45 @@ def __init__(self, context, request):
             },
         ]
 
-    def getInvoices(self, contentFilter):
-        return self.context.objectValues('Invoice')
-
-    # def __call__(self):
-    #     mtool = getToolByName(self.context, 'portal_membership')
-    #     addPortalMessage = self.context.plone_utils.addPortalMessage
-    #     if mtool.checkPermission(AddInvoice, self.context):
-    #         clients = self.context.clients.objectIds()
-    #         if clients:
-    #             self.context_actions[_('Add')] = {
-    #                 'url': 'createObject?type_name=Invoice',
-    #                 'icon': '++resource++bika.lims.images/add.png'
-    #             }
-    #     return super(InvoiceBatchInvoicesView, self).__call__()
-
-    def folderitems(self, full_objects=False):
+    def folderitem(self, obj, item, idx):
+        """
+        Replace or add the required/wanted fields for each invoice
+        in the item dictionary
+
+        :param obj: the instance of the class to be foldered. In our case, an
+                    Invoice
+        :param item: dict containing the properties of the object to be used by
+                     the template
+        :return: dictionary with the updated fields of the invoice being processed
+        """
         currency = currency_format(self.context, 'en')
-        self.show_all = True
-        self.contentsMethod = self.getInvoices
-        items = BikaListingView.folderitems(self, full_objects)
-        for item in items:
-            obj = item['obj']
-            number_link = "<a href='%s'>%s</a>" % (
-                item['url'], obj.getId()
+        number_link = "<a href='%s'>%s</a>" % (
+            item['url'], obj.getId()
+        )
+        item['replace']['id'] = number_link
+        if obj.getClient():
```

Comment:
Since you are using the client object in several places later, I suggest to use a local var to store the client:
```python
client = obj.getClient()
if client:
    [...]
```
and use it later instead of `obj.getClient()`